runtime/dart_isolate.cc

```diff
@@ -348,8 +348,15 @@ bool DartIsolate::PrepareForRunningFromKernel(
     return false;
   }
 
-  child_isolate_preparer_ = [mapping](DartIsolate* isolate) {
-    return isolate->PrepareForRunningFromKernel(mapping);
+  child_isolate_preparer_ = [this](DartIsolate* isolate) {
```

Comment:
Not sure this is thread safe. Instead of referring to this (which could have begun collection on a different thread immediately after spawning a new isolate). Can you can copy the vector and move it into the lambda? The copy is cheap since you are just copying a vector of pointers rather than the mappings themselves.

Reply:
So just `[kernel_buffers = kernel_buffers_](DartIsolate *isolate)` and only when `last_piece` is `true`.

Reply:
`last_piece` is already only true by this point. So just copy the kernel buffers vector and you should be good to go :)

Reply:
Sure, done.

Reply:
@chinmaygarde the test is failing with `[this]` to `[kernel_buffers = kernel_buffers_]` change. I am not sure about lifespan of that `kernel_buffers` vector that is captured by reference.